File: 프로그래머스/2/154539. 뒤에 있는 큰 수 찾기/뒤에 있는 큰 수 찾기.py
def solution(numbers):
    n = len(numbers)
    answer = [0 for _ in range(n)]
    stack=[0]
    for i in range(1,n):
        while stack and numbers[stack[-1]]<numbers[i]:
            answer[stack.pop()]=numbers[i]
        stack.append(i)
    while stack:
        answer[stack.pop()]=-1
        
    return answer

File: 프로그래머스/2/154539. 뒤에 있는 큰 수 찾기/뒤에 있는 큰 수 찾기.js
function solution(numbers) {
    var answer = [];
    let n = numbers.length
    
    for (let i=0;i<n;i++){
        answer.push(0)
    }

    let stack = [0]
    for(let i=1;i<n;i++){
        while (stack.length>0 &&numbers[stack[stack.length-1]]<numbers[i]){
            answer[stack.pop()]=numbers[i]
        }
            
        stack.push(i)
    }
    
    while (stack.length!=0){
        answer[stack.pop()]=-1
    }
    
    return answer;
}